cylc/flow/config.py

```diff
@@ -2346,6 +2346,23 @@ def load_graph(self):
             # dependencies are checked in generate_triggers:
             self.check_terminal_outputs(parser.terminals)
 
+        # set of all cycling intervals containined within the workflow
+        cycling_intervals = {
+            sequence.get_interval()
+            for sequence in self.sequences
+        } | {
+            # add a null interval to handle async workflows
+            get_interval_cls().get_null()
+        }
+
+        # determine the longest cycling interval in the workflow
+        self.interval_of_longest_sequence: 'IntervalBase' = max({
+            interval for interval in cycling_intervals
+            # NOTE: None type sorts above a null interval for strange
+            # historical reasons so must be filtered out
+            if interval is not None
+        })
```

Comment:
Is the implicit set creation needed?

Reply:
Good point, technically no, though it might be faster?

I suspect that `__hash__` comparisons are probably faster than `IntervalBase._cmp` comparisons, so filtering out duplicates (of which there will likely be many, e.g. `T00` and `T01` have the same interval of `P1D`) would make the `max` faster.

Don't much mind either way.